File: output/deepseek_chat/avatar/direct_translation/python/java/failed_translation/atcoder_ABC152_E.java
import java.util.*;
import java.math.*;

public class atcoder_ABC152_E {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int MOD = (int)1e9 + 7;
        int N = sc.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++) {
            A[i] = sc.nextInt();
        }
        long lcm = 1;
        for (int a : A) {
            lcm = lcm * a / gcd(lcm, a);
        }
        long sum = 0;
        for (int a : A) {
            sum = (sum + pow(a, MOD - 2, MOD)) % MOD;
        }
        long ans = (lcm % MOD) * sum % MOD;
        System.out.println((int)(ans % MOD));
    }

    private static long gcd(long a, long b) {
        return b == 0 ? a : gcd(b, a % b);
    }

    private static long pow(long a, long b, long mod) {
        long result = 1;
        a = a % mod;
        while (b > 0) {
            if (b % 2 == 1) {
                result = result * a % mod;
            }
            a = a * a % mod;
            b = b >> 1;
        }
        return result;
    }
}

File: dataset/avatar/Python/Code/atcoder_ABC152_E.py
from fractions import gcd
def lcm ( a , b ) :
    return a * b / gcd ( a , b )
MOD = 10 ** 9 + 7
N = int ( input ( ) )
A = list ( map ( int , input ( ).split ( ) ) )
_lcm = 1
for a in A :
    _lcm = _lcm * a // gcd ( _lcm , a )
ans = _lcm * sum ( [ pow ( s , MOD - 2 , MOD ) for s in A ] ) % MOD
print ( int ( ans % MOD ) )

